### DISCUSSION TITLE
Use drf ModelSerializer component to serialize PeriodicTask and set depth=1，create data report error

### DISCUSSION THREAD

**Discussion by slowchen** [celery/django-celery-beat#468] [Q&A]
### Summary:

Use drf ModelSerializer component to serialize PeriodicTask and set depth=1，create data report error django.core.exceptions.ValidationError: {'interval': ['One of clocked, interval, crontab, or solar must be set.']}

* Celery Version: 4.4.6
* Celery-Beat Version: 2.0.0
* Django Version: 3.0.6
* DRF Version: 3.11.0

### Exact steps to reproduce the issue:
1. create a `PeriodicTaskSerializer` by DRF Serializer
```python
class PeriodicTaskSerializer(serializers.ModelSerializer):
    class Meta:
        model = PeriodicTask
        fields = '__all__'
        depth = 1
```
2. create a `PeriodicTaskView` by DRF APIView,It accepts http post requests and can create tasks
```python
class PeriodicTaskView(APIView):
    """
    create periodic task
    """
    authentication_classes = ()

    def post(self, request, *args, **kwargs):
        request_data = request.data
        serializer = PeriodicTaskSerializer(data=request_data)
        if serializer.is_valid():
            serializer.save()
            return APIResponse(serializer.data)
        return APIResponse(data=serializer.errors)
```

3. Configure the url as follows And use postman to initiate the request
```json
{
    "name": "test create task",
    "task": "common.tasks.task_test",
    "interval": 1,
    "args": "[]",
    "kwargs": "{}",
    "queue": "",
    "priority": 1,
    "start_time": "2020-07-04 18:00:00",
    "one_off": "False",
    "enabled": "True",
    "description": "create task by drf"
}
```
![image](https://user-images.githubusercontent.com/17584002/86507469-0a99f100-be0b-11ea-87d5-d69b680d6b21.png)

### Detailed information
In the request parameters, I set the `interva=1`， but after `serializer.is_valid()`, there is no such parameter in the `serializer.data`, Cause an exception `{'interval': ['One of clocked, interval, crontab, or solar must be set.']}` to be thrown,
If I didn’t set the `depth` in the `PeriodicTaskSerializer`, everything works fine

**Comment by auvipy** [comment]
i think you should check the models well and set the input of interval. also you could try viewsets with hyperlinkedModel serializer and see if the CRUD operations are working properly
